src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/AccessRightService.java

```diff
@@ -270,58 +415,42 @@ public void deleteSoapServiceAccessRights(ClientId clientId, String fullServiceC
         return accessRightHolderDtos;
     }
 
-    /**
-     * Get matching {@link EndpointType endpoint} from {@link ClientType#endpoint client's list of endpoints}.
-     * @param clientType
-     * @param serviceType
-     * @param endpointMethod
-     * @param endpointPath
-     * @return
-     */
-    private Optional<EndpointType> getEndpoint(ClientType clientType, ServiceType serviceType, String endpointMethod,
-            String endpointPath) {
-        return clientType.getEndpoint().stream()
-                .filter(endpointType -> endpointType.getServiceCode().equals(serviceType.getServiceCode())
-                        && endpointType.getMethod().equals(endpointMethod)
-                        && endpointType.getPath().equals(endpointPath))
-                .findFirst();
-    }
-
-    /**
-     * Adds access rights to SOAP services. If the provided {@code subjectIds} do not exist in the serverconf db
-     * they will first be validated (that they exist in global conf) and then saved into the serverconf db.
-     * LocalGroup ids will also be verified and if they don't exist in the serverconf db they will be saved
-     * @param clientId
-     * @param fullServiceCode
-     * @param subjectIds
-     * @param localGroupIds
-     * @return List of {@link AccessRightHolderDto AccessRightHolderDtos}
-     * @throws LocalGroupNotFoundException
-     * @throws ClientNotFoundException
-     * @throws ServiceNotFoundException
-     * @throws EndpointNotFoundException
-     */
-    public List<AccessRightHolderDto> addSoapServiceAccessRights(ClientId clientId, String fullServiceCode,
-            Set<XRoadId> subjectIds, Set<Long> localGroupIds) throws LocalGroupNotFoundException,
-            ClientNotFoundException, ServiceNotFoundException, DuplicateAccessRightException,
-            IdentifierNotFoundException, EndpointNotFoundException {
+    private Set<XRoadId> mergeSubjectIdsWithLocalgroups(Set<XRoadId> subjectIds, Set<Long> localGroupIds)
+            throws IdentifierNotFoundException, LocalGroupNotFoundException {
         // Get persistent entities in order to change relations
         Set<XRoadId> txSubjects = new HashSet<>();
-        if (subjectIds != null) {
+        if (subjectIds != null && subjectIds.size() > 0) {
```

Comment:
nitpick: Sonar suggests "Use isEmpty() to check whether the collection is empty or not" https://sonarqube.niis.org/project/issues?id=xroad&issues=AXDETU4aBW8O1Uq1Fo6y&open=AXDETU4aBW8O1Uq1Fo6y&pullRequest=408

Reply:
Fixed.